src/vs/editor/browser/widget/diffEditorWidget2/utils.ts

```diff
@@ -170,3 +170,19 @@ export function animatedObservable(base: IObservable<number, boolean>, store: Di
 function easeOutExpo(t: number, b: number, c: number, d: number): number {
 	return t === d ? b + c : c * (-Math.pow(2, -10 * t / d) + 1) + b;
 }
+
+export function deepMerge<T extends {}>(source1: T, source2: Partial<T>): T {
```

Comment:
This seems similar to https://github.com/microsoft/vscode/blob/main/src/vs/base/common/objects.ts#L89

Maybe use that or a deepClone and then mixin if you want a truly new object.

Reply:
mixin uses `any` types...